clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -17556,11 +17584,12 @@ Decl *Sema::ActOnFriendTypeDecl(Scope *S, const DeclSpec &DS,
 
   Decl *D;
   if (!TempParams.empty())
+    // TODO: Support variadic friend template decls?
```

Comment:
Can we create an issue for that?

Reply:
I can do that yeah. We just generally don’t support this type of friend declaration, irrespective of whether it’s a pack expansion or not. I can open an issue for that in general if there isn’t already one, because once that’s implemented, supporting pack expansions should be straight-forward.

Reply:
I’ve opened an issue for supporting friend declarations w/ dependent NNSs: #104057

Once we actually start using `FriendTemplateDecl`s properly (which from what I can tell is what’s required for that), then this should be a natural follow-up.